src/main/java/com/elharo/propernames/Names.java

```diff
@@ -0,0 +1,89 @@
+package com.elharo.propernames;
+
+/**
+ * Utility class to check whether a string is likely to be a proper name.
+ */
+public class Names {
+    
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private Names() {
+        // Utility class - prevent instantiation
+    }
+    
+    /**
+     * Checks whether the given string is likely to be a proper name.
+     *
+     * @param name the string to check
+     * @return true if the string is likely to be a proper name, false otherwise
+     */
+    public static boolean isProperName(String name) {
+        if (name == null || name.trim().isEmpty()) {
+            return false;
+        }
+        
+        String trimmed = name.trim();
+        if (trimmed.length() == 0) {
+            return false;
+        }
+        
+        // Split by spaces to check each word
+        String[] words = trimmed.split("\\s+");
+        
+        for (String word : words) {
+            if (!isValidProperNameWord(word)) {
+                return false;
+            }
+        }
+        
+        return true;
+    }
+    
+    /**
+     * Checks if a single word is a valid proper name word.
+     * A valid proper name word should:
+     * - Start with an uppercase letter
+     * - Have remaining letters in lowercase (except for special cases like O'Connor)
+     * - Only contain letters, apostrophes, periods, and hyphens
```

Comment:
proper names do not contain punctuation except for apostrophe, hyphen, and single quote